### PR TITLE
SW-1852 add latest-changes-for-grblhal-to-repo

### PR DESCRIPTION
**Pull request by amrithvenkat**
Create a PR to accomodate changes done by Basti

### CODE REVIEW

File: CMakeLists.org [hunk 1]
```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.12)
```
Comment (ahmed-mrbeam) [inline CMakeLists.org]: We don't need this file, this is a copy of the original file and if we want to refer to the original file later, we can see it in the history or refer to the original repo we forked from. Please remove

File: mrbeam_picogrbl_map.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
+/*
```
Comment (ahmed-mrbeam) [inline mrbeam_picogrbl_map.h:1]: Header file Guards are missing for this file. Kindly add

Comment (amrithvenkat) [reply mrbeam_picogrbl_map.h:1]: Implemented

File: CMakeLists.org (hunk 1 above)
Comment (amrithvenkat) [reply CMakeLists.org]: Removed the file

**Pull request closed without merging**